packages/runtime/container-runtime/src/summary/runningSummarizer.ts

```diff
@@ -243,12 +243,20 @@ export class RunningSummarizer extends TypedEventEmitter<ISummarizerEvents> impl
 			}
 		});
 
+		const immediatelyRefreshLatestSummaryAck =
+			this.mc.config.getBoolean("Fluid.Summarizer.immediatelyRefreshLatestSummaryAck") ??
+			false;
 		this.generator = new SummaryGenerator(
 			this.pendingAckTimer,
 			this.heuristicData,
 			this.submitSummaryCallback,
-			() => {
+			async (options: IRefreshSummaryAckOptions) => {
 				this.totalSuccessfulAttempts++;
+				if (immediatelyRefreshLatestSummaryAck) {
```

Comment:
What are the pros / cons of doing this in SummaryGenerator vs in the `afterSumaryAction` of a `trySummarize` (the solution we discussed in the offline conversation).

Reply:
I think `this.generator.summarize` has several entry-points such as `trySummarize`, `trySummarizeOnce`, `trySummarizeWithRetries` and `trySummarizeWithStaticAttempts`; so by injecting this code in `successfulSummaryCallback` we centralize the fix as opposed to doing it only in `trySummarize` where the other summarizing code-paths may not get covered. 

One could argue that this snippet could also be added to `this.afterSummaryAction()` as is, but it's easier to miss out if a new entry point to  `this.generator.summarize` down the line by a newer engineer. So adding it right at the initialization step of `SummaryGenerator`, ensures more safety against that. Just my thoughts, Tyler would know more.

Reply:
The afterSummaryAction method is synchronous, and it would need to be made async. That has its first host of complications. Rajatch's point here is quite succint. The third is that the Summary Generator already waits for the ack, and it feels like it should be responsible for generating and acking the summary. It is already doing this async, so it's easier to add the await there as well.